src/ui/public/vis/editors/default/sidebar.html

```diff
@@ -75,6 +76,23 @@
             </div>
           </li>
 
+          <li
+            tooltip="Auto apply changes"
+            tooltip-placement="bottom"
+            tooltip-popup-delay="400" tooltip-append-to-body="1"
+            ng-if="vis.type.editorConfig.enableAutoApply"
+          >
+            <button
+              data-test-subj="visualizeEditorAutoButton"
+              class="kuiButton kuiButton--basic navbar-btn-link"
+              aria-label="Auto update the visualization on every changes"
+              ng-click="toggleAutoApply()"
```

Comment:
I would use `ng-mousedown="toggleAutoApply();$event.preventDefault()"` instead of ng-click here - prevents focus loss.  It would be great if the form submit (play button) would also do that, but that's trickier. The cancel button probably shouldn't keep focus because form content may changen.

Reply:
We should not do this, due to destroying the default behavior of buttons, which would be an accessibility issue.

Instead Peter just implemented pressing <kbd>Ctrl</kbd> + <kbd>Enter</kbd> to submit the whole editor, so you never have to use the mouse to click this button, and can type and apply without leaving the actual input fields.